### ISSUE TITLE
Github: generalise to notifications API

### ISSUE THREAD

**Issue by OJFord** [wtfutil/wtf#898]
## What should it do?
I'd hoped the `github` module was going to give me GH notifications, rather than be sort of it's own specific notifications implementation on top of the PRs API, for specific repos.

The current behaviour would still be achievable through filtering if it were made more generic, to get all notifications (issue comments, merges, etc.).

(Note this is unrelated to #803 which is about actual system notifications, I mean just listing the same 'notifications' that GitHub shows in the browser when you click the bell in the top-right.)

**Comment by senorprogrammer** [comment]
As is, this isn't actionable as a bug or feature request. This is just an existing capability not meeting your expectations. I'm more than happy to accept a PR that provides a GitHub Notifications module, or extends the existing one to support notifications.

**Comment by OJFord** [comment]
@senorprogrammer I don't think I phrased it very well - the issue is that what I'd like is something like 'assigned to me, in any repo', but currently the module requires a list of repositories to check, so as far as I can tell this isn't possible.

I focussed a bit too much I think on the solution being to use `/notifications` instead of `/repos`, I guess it would be possible with the latter anyway (just by running through the list), just less efficient.

The other types of notifications that `/notifications` offers that I don't think are available from `/repos` would just be nice to have, main thing is that I don't want to filter on specific repos.

I rarely use Go so it might be a bit much for me, but I might give it a shot if you're not against it in principle.

Cheers.
